lib/import_external_activity.rb

```diff
@@ -0,0 +1,48 @@
+class ImportExternalActivity < Struct.new(:import,:activity_json,:portal_url,:current_visitor_id)
+
+  def perform
+    uri = URI.parse("#{APP_CONFIG[:authoring_site_url]}/import/import_portal_activity")
+
+    begin
+      Timeout.timeout(60) {
+        client = Client.find(:first, :conditions => {:domain => APP_CONFIG[:authoring_site_url]})
+        auth_token = 'Bearer %s' % client.app_secret
+      	response = HTTParty.post(uri.to_s,
+	      :body => {
+	        :portal_url => portal_url,
+	        :activity => activity_json,
+	        :domain_uid => current_visitor_id
+          }.to_json,
+	      :headers => {"Content-Type" => 'application/json', "Authorization" => auth_token})
+
+	    if response.code == 200 #successful
+	      activity_data = JSON.parse response.headers['data'], :symbolize_names => true
+	      if activity_data[:response_code] == 201
+	        import.update_attribute(:job_finished_at, Time.current)
+	        import.update_attribute(:progress, 100)
+	        import_activity = ExternalActivity.find(activity_data[:external_activity_id])
+	        import_activity.cohort_list = activity_json[:cohort_list]
+            import_activity.publication_status = activity_json[:publication_status].nil? ? "published" : activity_json[:publication_status] == "published" ? "published" : "private"
+	        import_activity.save
+            #give author role to creator of activity
+	        user = User.find_by_email(activity_json[:user_email])
+	        user.add_role("author") if user
```

Comment:
@scytacki Quick link to checkout the changes for this commit https://github.com/concord-consortium/rigse/pull/93/files#diff-cae80054481fbe8d1dcf41e37090c87eR30